spring-cloud-data-rest-client/src/main/java/org/springframework/cloud/data/rest/client/CloudDataTemplate.java

```diff
@@ -55,10 +60,16 @@ public CloudDataTemplate(URI baseURI) {
 		resources.put("streams/definitions", new UriTemplate(resourceSupport.getLink("streams").getHref() + "/definitions"));
 		resources.put("streams/deployments", new UriTemplate(resourceSupport.getLink("streams").getHref() + "/deployments"));
 		this.streamTemplate = new StreamTemplate(restTemplate, resources);
+		this.counterTemplate = new CounterTemplate(restTemplate, resourceSupport);
```

Comment:
In any case, the `resources` variable above is a field of the `CloudDataTemplate`, so I was merely trying to use a container which seemed more appropriate (collection of links). Also, we'd need to address that concatenation thing done for Streams (eg by splitting definitions and deployments concerns at the controller level).

Can we use jdk8 in this code? stream filter would come handy here...